python/ray/tune/tune.py

```diff
@@ -698,22 +677,25 @@ class and registered trainables.
         local_path=local_path, remote_path=remote_path, sync_config=sync_config
     )
 
-    checkpoint_score_attr = checkpoint_score_attr or ""
+    checkpoint_config = checkpoint_config or CheckpointConfig()
+    checkpoint_score_attr = (
+        # Note(jungong) : we must use _tune_legacy_checkpoint_score_attr
+        # here, instead of checkpoint_score_attribute, to make sure we
+        # handle legacy cases correctly (e.g., someone purposefully
+        # specify checkpoint_score_order="min-loss").
+        checkpoint_config._tune_legacy_checkpoint_score_attr
+        or ""
+    )
     if checkpoint_score_attr.startswith("min-"):
-        checkpoint_score_attr = checkpoint_score_attr[4:]
-        checkpoint_score_order = "min"
+        logger.warning(
+            "using min- and max- prefixes to specify checkpoint score "
+            "order is deprecated. Please use CheckpointConfig.checkpoint_score_order "
+            "instead"
+        )
+        checkpoint_config.checkpoint_score_attribute = checkpoint_score_attr[4:]
+        checkpoint_config.checkpoint_score_order = "min"
     else:
-        checkpoint_score_order = "max"
-
-    checkpoint_config = CheckpointConfig(
-        num_to_keep=keep_checkpoints_num,
-        checkpoint_score_attribute=checkpoint_score_attr,
-        checkpoint_score_order=checkpoint_score_order,
-        checkpoint_frequency=checkpoint_freq,
-        checkpoint_at_end=checkpoint_at_end,
-        _checkpoint_keep_all_ranks=checkpoint_keep_all_ranks,
-        _checkpoint_upload_from_workers=checkpoint_upload_from_workers,
-    )
+        checkpoint_config.checkpoint_score_order = "max"
```

Comment:
Does this overwrite the user's request with max always? I think we should get rid of this else case.

Like if they want:

```
checkpoint_score_attribute = "loss"
checkpoint_score_order = "min" ->  gets changed to "max" since there's no "min-" prefix?
```

Reply:
checkout ``_tune_legacy_checkpoint_score_attr()``, this is the intended behavior actually ...
if we have min-xxx, that is to minimize xxx, otherwise we maximize xxx, including if the field is just "xxx".
I don't know how this was designed ...

Reply:
I think we still have to do

```
    elif checkpoint_score_attr:
        checkpoint_config.checkpoint_score_attribute = checkpoint_score_attr
        checkpoint_config.checkpoint_score_order = "max"
```

Reply:
If a user does:

```python
config = CheckpointConfig(
    checkpoint_score_attribute="min-training_iteration",
    checkpoint_score_order="max",
)

config._tune_legacy_checkpoint_score_attr --> "min-min-training_iteration"
```

Then, we'll be trying to minimize `min-training_iteration` which doesn't exist.

@gjoliver  Can we fix + add a test for that?

Reply:
Ah, I see this is why we do `else: "max"`:


```python
config = CheckpointConfig(
    checkpoint_score_attribute="loss",
    checkpoint_score_order="min",
)

config._tune_legacy_checkpoint_score_attr --> "min-loss"
```

Reply:
I think we should deprecate this `min-` behavior while we're at it.